### ISSUE TITLE
Quay database crashes after OpenShift upgrade

### ISSUE THREAD

**Issue by kevchu3** [quay/quay-operator#230]
I have Quay Operator and QuayEcosystem CR deployed into my quay-enterprise namespace, and am using the local storage operator to provide RWO persistent storage.  I've performed about a dozen successful OpenShift minor version upgrades in the past few months (successful from the cluster operator perspective).  However, each time my example-quayecosystem-quay pod encounters a CrashLoopBackOff with the same error, while the remaining pods are still running.  I figure to file a bug on this as I've been able to reproduce it consistently, and suspect this has something to do with the persistent storage.  As a workaround, I have to delete and redeploy the QuayEcosystem CR which essentially cleans up the persistent storage so I can start again (not good for persistence).

My quay-enterprise namespace:
```
[8:49:33] root:~ # oc get pods  
NAME                                                    READY   STATUS             RESTARTS   AGE
example-quayecosystem-quay-7c586fb57f-qkgjl             0/1     CrashLoopBackOff   3          2m
example-quayecosystem-quay-postgresql-9dc844568-scpn8   1/1     Running            0          4d21h
example-quayecosystem-redis-5f8bd6bfd4-xk8qv            1/1     Running            0          4d21h
quay-operator-7f98967f87-vrnz4                          1/1     Running            0          4d21h
```

My QuayEcosystem CR:
```
apiVersion: redhatcop.redhat.io/v1alpha1
kind: QuayEcosystem
metadata:
  name: example-quayecosystem
  namespace: quay-enterprise
spec:
  quay:
    imagePullSecretName: redhat-quay-pull-secret
    registryStorage:
      persistentVolumeAccessModes:
        - ReadWriteOnce
      persistentVolumeSize: 5Gi
      persistentVolumeStorageClassName: local-storageclass
```

The SQL error after each OpenShift upgrade appears to be consistent:
```
12:43:48 INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
12:43:48 INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
12:43:48 INFO  [alembic.runtime.migration] Running upgrade d42c175b439a -> e2894a3a3c19, Add full text search indexing for repo name and description
Traceback (most recent call last):
  File "/opt/rh/python27/root/usr/bin/alembic", line 8, in <module>
    sys.exit(main())
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/alembic/config.py", line 540, in main
    CommandLine(prog=prog).main(argv=argv)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/alembic/config.py", line 534, in main
    self.run_cmd(cfg, options)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/alembic/config.py", line 514, in run_cmd
    **dict((k, getattr(options, k, None)) for k in kwarg)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/alembic/command.py", line 279, in upgrade
    script.run_env()
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/alembic/script/base.py", line 475, in run_env
    util.load_python_file(self.dir, "env.py")
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/alembic/util/pyfiles.py", line 98, in load_python_file
    module = load_module_py(module_id, path)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/alembic/util/compat.py", line 240, in load_module_py
    mod = imp.load_source(module_id, path, fp)
  File "data/migrations/env.py", line 170, in <module>
    run_migrations_online()
  File "data/migrations/env.py", line 146, in run_migrations_online
    tables=tables, tester=get_tester(), progress_reporter=progress_reporter
  File "<string>", line 8, in run_migrations
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/alembic/runtime/environment.py", line 846, in run_migrations
    self.get_context().run_migrations(**kw)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/alembic/runtime/migration.py", line 365, in run_migrations
    step.migration_fn(**kw)
  File "/quay-registry/data/migrations/versions/e2894a3a3c19_add_full_text_search_indexing_for_repo_.py", line 29, in upgrade
    mysql_prefix="FULLTEXT",
  File "/quay-registry/data/migrations/progress.py", line 98, in wrapped_method
    result = maybe_callable(*args, **kwargs)
  File "<string>", line 8, in create_index
  File "<string>", line 3, in create_index
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/alembic/operations/ops.py", line 975, in create_index
    return operations.invoke(op)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/alembic/operations/base.py", line 345, in invoke
    return fn(self, operation)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/alembic/operations/toimpl.py", line 88, in create_index
    operations.impl.create_index(idx)
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/alembic/ddl/impl.py", line 276, in create_index
    self._exec(schema.CreateIndex(index))
  File "/opt/rh/python27/root/usr/lib/python2.7/site-packages/alembic/ddl/impl.py", line 134, in _exec
    return conn.execute(construct, *multiparams, **params)
  File "/opt/rh/python27/root/usr/lib64/python2.7/site-packages/sqlalchemy/engine/base.py", line 988, in execute
    return meth(self, multiparams, params)
  File "/opt/rh/python27/root/usr/lib64/python2.7/site-packages/sqlalchemy/sql/ddl.py", line 72, in _execute_on_connection
    return connection._execute_ddl(self, multiparams, params)
  File "/opt/rh/python27/root/usr/lib64/python2.7/site-packages/sqlalchemy/engine/base.py", line 1050, in _execute_ddl
    compiled,
  File "/opt/rh/python27/root/usr/lib64/python2.7/site-packages/sqlalchemy/engine/base.py", line 1253, in _execute_context
    e, statement, parameters, cursor, context
  File "/opt/rh/python27/root/usr/lib64/python2.7/site-packages/sqlalchemy/engine/base.py", line 1473, in _handle_dbapi_exception
    util.raise_from_cause(sqlalchemy_exception, exc_info)
  File "/opt/rh/python27/root/usr/lib64/python2.7/site-packages/sqlalchemy/util/compat.py", line 398, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/opt/rh/python27/root/usr/lib64/python2.7/site-packages/sqlalchemy/engine/base.py", line 1249, in _execute_context
    cursor, statement, parameters, context
  File "/opt/rh/python27/root/usr/lib64/python2.7/site-packages/sqlalchemy/engine/default.py", line 552, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedObject) operator class "gin_trgm_ops" does not exist for access method "gin"

[SQL: CREATE INDEX repository_description__fulltext ON repository USING gin (description gin_trgm_ops)]
(Background on this error at: http://sqlalche.me/e/f405)
```

Reproduced by:
* Applying OpenShift minor version upgrade
* QuayEcosystem CR is deployed on persistent storage, with an RWO PVC managed by the local storage operator

Versions tested:
* OpenShift 4.3.x to 4.4.x
* Quay v1.0.2 (stable channel) from OperatorHub

**Comment by HenriAW** [comment]
I am also having the same issue. Were you able to resolve this?

**Comment by kevchu3** [comment]
@HenriAW I don't have my original Quay Operator deployed anymore, but it's also worth noting that the Quay Operator is now up to v3.3, in case the newer version might resolve your issue.

**Comment by HenriAW** [comment]
@kevchu3, thanks for your response. I am using the v3.3 operator and am hitting the same issue so it looks to still be a bug, I'll let you know if I'm able to figure out the cause

**Comment by ricardomaraschini** [comment]
@HenriAW Can you replicate the issue with the last version (v3.6.2) ?

**Comment by rh-tguittet** [comment]
I ran into the very same issue when running `make local-dev-up` in v3.6.2 for quay. Here is my awfully dirty fix for it:
```patch
diff --git a/data/migrations/versions/e2894a3a3c19_add_full_text_search_indexing_for_repo_.py b/data/migrations/versions/e2894a3a3c19_add_full_text_search_indexing_for_repo_.py
index b616fbd6..0c482207 100644
--- a/data/migrations/versions/e2894a3a3c19_add_full_text_search_indexing_for_repo_.py
+++ b/data/migrations/versions/e2894a3a3c19_add_full_text_search_indexing_for_repo_.py
@@ -16,6 +16,7 @@ from sqlalchemy.dialects import mysql
 
 def upgrade(op, tables, tester):
     # ### commands auto generated by Alembic - please adjust! ###
+    op.execute(sa.text("CREATE EXTENSION IF NOT EXISTS pg_trgm;"))
     op.create_index(
         "repository_description__fulltext",
         "repository",
```

**Comment by ricardomaraschini** [comment]
This is fixed in v3.7.0 (not released yet).
